packages/@aws-cdk/aws-dynamodb/lib/table.ts

```diff
@@ -23,6 +23,12 @@ export interface TableProps {
      * @default <generated>
      */
     tableName?: string;
+
+    /**
+     * The Type of Stream to create.
+     * @default ''
+     */
+    streamSpecification?: cloudformation.TableResource.StreamSpecificationProperty;
```

Comment:
We usually don't like to expose the raw CloudFormation data types in this layer since normally there's a way to provide a richer API. This is a great example for this. the [StreamSpecification] property basically has a single string field that can accept a bunch of possible values. Instead of sending users to read the docs and find what are the allowed values (failing only in deploy-time), we can just provide a simple enum:

```
export enum StreamViewType {
  NewImage = 'NEW_IMAGE',
  OldImage = 'OLD_IMAGE',
  NewAndOldImages = 'NEW_AND_OLD_IMAGES',
  KeysOnly = 'KEYS_ONLY'
}
```

Also, copy & paste the documentation from the [DynamoDB manual](https://docs.aws.amazon.com/amazondynamodb/latest/APIReference/API_StreamSpecification.html) as jsdocs on these enum members so users have all the information.

I also don't see any need for a nested structure here. We can use the fact that this is an optional field to determine if streams are enabled. So you can just call this `streamViewType` and indicate under `@default` that if this is not defined, DDB streams are disabled.